src/redux/books/books.js

```diff
@@ -1,22 +1,49 @@
+import { addNewBook, fetchBooks, deleteBook } from '../../components/Api';
+
 const ADD_BOOK = 'bookStore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
+const FETCH_BOOKS = 'bookStore/books/FETCH_BOOKS';
+
 const initialState = [];
-export const addBook = (payload) => ({
-  type: ADD_BOOK,
-  payload,
-});
+export const addBook = (newBook) => async (dispatch) => {
+  await addNewBook(newBook);
+  dispatch({
+    type: ADD_BOOK,
+    payload: newBook,
+  });
+};
 
-export const removeBook = (payload) => ({
-  type: REMOVE_BOOK,
-  payload,
-});
+export const removeBook = (payload) => async (dispatch) => {
+  await deleteBook(payload.id);
+  dispatch({
+    type: REMOVE_BOOK,
+    payload,
+  });
+};
+
+export const fetchBook = () => async (dispatch) => {
+  const book = await fetchBooks();
+  const bookList = Object.entries(book).map(([key, value]) => {
+    const [bookEntries] = value;
+    return {
+      id: key,
+      ...bookEntries,
+    };
+  });
+  dispatch({
+    type: FETCH_BOOKS,
+    payload: bookList,
+  });
```

Comment:
- [ ] Good job on  adding, deleting, and fetching your books to the API but when rendering your books there is a bit of issue, When I go to the categories page it works fine I see the message `under construction but when I navigate back to books they duplicate themselves but if I reload the page the duplication disappears, Kindly have a look at that video and fix the issue :point_down: 

https://user-images.githubusercontent.com/75126481/151522207-21c1c5cc-13cd-4e61-843f-7b6547dde398.MP4

Reply:
Hello @hiromataba ,

Thanks a million for spotting the issue 😄 . 

I have successfully managed to fix the issue and will on requesting for a final review.

> HappyCoding 🤓

Reply:
Thank you @blessedjasonmwanza :pray: